components/screens/full-pages/DeployingToken.tsx

```diff
@@ -0,0 +1,117 @@
+import { useNavigate } from "react-router-dom";
+import React, { useEffect, useState } from "react";
+import { useLocation } from "react-router";
+import Header from "@/components/GeneralHeader.tsx";
+import { deploy, ForboleFee } from "@/lib/krc20.ts";
+import carriageImage from "@/assets/images/carriage.png";
+import { kaspaToSompi } from "@/wasm/core/kaspa";
+import { FORBOLE_PAYOUT_ADDRESSES } from "@/lib/forbole.ts";
+import { NetworkType } from "@/contexts/SettingsContext.tsx";
+import { WalletSecret } from "@/types/WalletSecret.ts";
+import { AccountFactory } from "@/lib/wallet/wallet-factory.ts";
+import { DeployTokenState } from "@/components/screens/full-pages/DeployToken.tsx";
+
+export default function DeployingToken() {
+  const navigate = useNavigate();
+  const calledOnce = useRef(false);
+
+  const { state } = useLocation();
+  const { ticker, maxSupply, mintAmount, preAllocation, decimalPlaces } =
+    state as DeployTokenState;
+
+  const { rpcClient, networkId = NetworkType.Mainnet } = useRpcClientStateful();
+  const [secret, setSecret] = useState<WalletSecret>();
+  const { getWalletSecret } = useKeyring();
+  const { walletSettings } = useWalletManager();
+
+  useEffect(() => {
+    if (!walletSettings?.selectedWalletId) {
+      return;
+    }
+
+    getWalletSecret({
+      walletId: walletSettings.selectedWalletId,
+    }).then(({ walletSecret }) => setSecret(walletSecret));
+  }, [walletSettings]);
+
+  const accountFactory = !rpcClient
+    ? undefined
+    : new AccountFactory(rpcClient, networkId);
+
+  useEffect(() => {
+    if (
+      !rpcClient ||
+      walletSettings?.selectedAccountIndex === undefined ||
+      walletSettings?.selectedAccountIndex === null ||
+      !secret ||
+      !accountFactory
+    )
+      return;
+
+    const broadcastOperation = async () => {
+      const account =
+        secret.type === "mnemonic"
+          ? accountFactory.createFromMnemonic(
+              secret.value,
+              walletSettings.selectedAccountIndex,
+            )
+          : accountFactory.createFromPrivateKey(secret.value);
+
+      for await (const _step of deploy(
```

Comment:
Honestly, I don't see advantage of applying yield iterator here. It is the same use just call `await deploy(...)`